web/integration-core/src/jvmTest/kotlin/org/jetbrains/compose/web/tests/integration/EventTests.kt

```diff
@@ -236,4 +254,151 @@ class EventTests : BaseIntegrationTests() {
+        val textToCopy1 = driver.findElement(By.id("txt_to_copy"))
+        val textinput1 = driver.findElement(By.id("textinput"))
+
+        driver.copyElementTextToClipboard(textToCopy1)
+        driver.pasteFromClipboardInto(textinput1)
+
+        // Copied text should be changed by the onCopy handler
+        driver.waitTextToBe(value = "COPIED_TEXT_WAS_OVERRIDDEN".lowercase())
+    }
+
+    @ResolveDrivers
+    fun copyPasteUnchangedEventsTest(driver: WebDriver) {
+        driver.openTestPage("copyPasteEventsTest")
+        driver.waitTextToBe(value = "None")
+
+        val textinput2 = driver.findElement(By.id("textinput"))
+        val textToCopy2 = driver.findElement(By.id("txt_to_copy2"))
+
+        driver.copyElementTextToClipboard(textToCopy2)
+        driver.pasteFromClipboardInto(textinput2)
+
+        // Copied text should not be changed by the onCopy handler
+        driver.waitTextToBe(value = "SomeTestTextToCopy2".lowercase())
+    }
+
+    @ResolveDrivers
+    fun cutPasteEventsTests(driver: WebDriver) {
+        driver.openTestPage("cutPasteEventsTest")
+        driver.waitTextToBe(value = "None")
+
+        val textinput1 = driver.findElement(By.id("textinput1"))
+        val textinput2 = driver.findElement(By.id("textinput2"))
+
+        driver.cutElementTextToClipboard(textinput1)
+        driver.waitTextToBe(value = "Text was cut")
+
+        driver.pasteFromClipboardInto(textinput2)
+        driver.waitTextToBe(value = "Modified pasted text = TextToCut")
+    }
+
+    @ResolveDrivers
+    fun keyDownKeyUpTest(driver: WebDriver) {
+        driver.openTestPage("keyDownKeyUpTest")
+
+        driver.waitTextToBe(value = "None", textId = "txt_down")
+        driver.waitTextToBe(value = "None", textId = "txt_up")
+
+        val input = driver.findElement(By.id("textinput"))
+        input.sendKeys("a")
+
+        driver.waitTextToBe(value = "keydown = a", textId = "txt_down")
+        driver.waitTextToBe(value = "keyup = a", textId = "txt_up")
+
+        Actions(driver).keyDown(input, Keys.SHIFT).perform()
+        driver.waitTextToBe(value = "keydown = Shift", textId = "txt_down")
+        driver.waitTextToBe(value = "keyup = a", textId = "txt_up")
+
+        Actions(driver).keyUp(input, Keys.SHIFT).perform()
+        driver.waitTextToBe(value = "keydown = Shift", textId = "txt_down")
+        driver.waitTextToBe(value = "keyup = Shift", textId = "txt_up")
+    }
+
+    //@ResolveDrivers
+    // TODO: at least chrome driver has mobile emulation, so we can try it out (didn't work right away)
+    fun touchEventsDispatched(driver: WebDriver) {
+        driver.openTestPage("touchEventsDispatched")
+        driver.waitTextToBe(value = "None", textId = "txt_start")
+        driver.waitTextToBe(value = "None", textId = "txt_move")
+        driver.waitTextToBe(value = "None", textId = "txt_end")
+
+        val box = driver.findElement(By.id("box"))
+
+        TouchActions(driver)
+            .down(box.rect.x + 50, box.rect.y + 50)
```

Comment:
may it we can just check sizes and click sort of in the middle (or wherever we want to click)

Reply:
fixed!